src/components/TVChartContainer/TVChartContainer.tsx

```diff
@@ -0,0 +1,172 @@
+        localStorage.setItem(TV_CHART_RELOAD_TIMESTAMP_KEY, Date.now().toString());
+      } else {
+        const tvReloadTimestamp = Number(localStorage.getItem(TV_CHART_RELOAD_TIMESTAMP_KEY));
+        if (tvReloadTimestamp && Date.now() - tvReloadTimestamp > TV_RELOAD_INTERVAL) {
+          if (resetCache.current) {
+            resetCache.current();
+            tvWidgetRef.current?.activeChart().resetData();
+          }
+        }
+      }
+    };
+
+    document.addEventListener("visibilitychange", handleVisibilityChange);
+
+    return () => {
+      document.removeEventListener("visibilitychange", handleVisibilityChange);
+    };
+  }, [resetCache]);
+
+  useEffect(
+    function updateLines() {
+      const lines: (IPositionLineAdapter | undefined)[] = [];
+      if (savedShouldShowPositionLines) {
+        chartLines.forEach((order) => {
+          lines.push(drawLineOnChart(order.title, order.price));
+        });
+      }
+      return () => {
+        lines.forEach((line) => line?.remove());
+      };
+    },
+    [chartLines, savedShouldShowPositionLines, drawLineOnChart]
+  );
+
+  useEffect(() => {
+    if (chartReady && tvWidgetRef.current && symbol !== tvWidgetRef.current?.activeChart?.().symbol()) {
+      tvWidgetRef.current.setSymbol(symbol, tvWidgetRef.current.activeChart().resolution(), () => {});
+    }
+  }, [symbol, chartReady, period]);
+
+  useEffect(() => {
+    if (!chartReady || !tvWidgetRef.current || !isMobile) return;
+
+    if (tvWidgetRef.current.activeChart().getCheckableActionState("drawingToolbarAction")) {
+      tvWidgetRef.current?.activeChart().executeActionById("drawingToolbarAction");
+    }
+  }, [isMobile, chartReady]);
+
+  useEffect(() => {
+    const widgetOptions = {
+      debug: false,
+      symbol: symbolRef.current, // Using ref to avoid unnecessary re-renders on symbol change and still have access to the latest symbol
+      datafeed: datafeed,
+      theme: defaultChartProps.theme,
+      container: chartContainerRef.current,
+      library_path: defaultChartProps.library_path,
+      locale: defaultChartProps.locale,
+      loading_screen: defaultChartProps.loading_screen,
+      enabled_features: defaultChartProps.enabled_features,
+      disabled_features: isMobile
+        ? defaultChartProps.disabled_features.concat(disabledFeaturesOnMobile)
+        : defaultChartProps.disabled_features,
+      client_id: defaultChartProps.clientId,
+      user_id: defaultChartProps.userId,
+      fullscreen: defaultChartProps.fullscreen,
+      autosize: defaultChartProps.autosize,
+      custom_css_url: defaultChartProps.custom_css_url,
+      overrides: defaultChartProps.overrides,
+      interval: getObjectKeyFromValue(period, SUPPORTED_RESOLUTIONS),
+      save_load_adapter: new SaveLoadAdapter(chainId, tvCharts, setTvCharts, onSelectToken),
+    };
+    tvWidgetRef.current = new window.TradingView.widget(widgetOptions);
+    tvWidgetRef.current?.onChartReady(function () {
```

Comment:
`current` is always defined here, right? if so you can use `!` instead of `?` to tell typescript that value is defined here